### PR TITLE
Implement pressure gradient as a source instead of a divergence to pressure hydrostatic balance

### PR DESCRIPTION
**Pull request by blallen**
### Description

Does what the title says for the ThreeDimensionalCompressibleNavierStokesEquations. Preserves hydrostatic balance out to grid-level precision (1e-10). 

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [ ] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [ ] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by blallen** [comment]
bors try

**Review by jkozdon — APPROVED** [review]

File: test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl [hunk 1]
```diff
@@ -101,6 +101,8 @@ ClimateMachine.init()
         for (key2, orientation) in orientations
             key = key1 * key2
 
+            println("running ", key)
```
Comment (jakebolewski) [inline test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:104]: leftover debug commet

Comment (blallen) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:104]: this is actually so it's easy to check which state check test is which during the CI

**Comment by blallen** [comment]
bors try

**Pull request merged**